src/main/java/com/aws/greengrass/deployment/DeviceConfiguration.java

```diff
@@ -709,10 +709,11 @@ public boolean isDeviceConfiguredToTalkToCloud() {
         try {
             validate(true);
             deviceConfigValidateCachedResult.set(true);
```

Comment:
Make it AtomicBoolean? Right now it's `AtomicReference<Boolean>`

Reply:
No we don't want that. This is a tri-state: unset, set true, set false.